spec/index.md

```diff
@@ -2301,6 +2301,34 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
 The [standard nearest-rank estimation method](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method), inclusive, with the addition of a 0th percentile is used. Concretely, for any i from 1 to 100, the ith percentile is the fee with rank `⌈i * 100⌉`. The 0th percentile is defined as the smallest fee (excluding coinbase transactions).
 
+### IC method `take_snapshot` {#ic-take_snapshot}
+
+This method takes a snapshot of the canister. Subsequent `take_snapshot` calls will replace the existing snapshot with a new one.
+
+The optional `duration` parameter can be used to limit the storage duration of a snapshot. When provided, the snapshot will be stored for the specified duration (in seconds) and will be deleted automatically at the end of this duration. In this case,
+subsequent calls to `load_snapshot` or `delete_snapshot` will fail.  
+
+It's important to note that a snapshot will increase the memory footprint of the canister. Thus, the balance must have the right amount of cycles to support the new freezing threshold.
+
+Only controllers can take a snapshot of a canister and restore it.
+
+### IC method `load_snapshot` {#ic-take_snapshot}
+
+This method loads a snapshot identified by `snapshot_id` onto the canister.  
+
+Only controllers can take a snapshot of a canister and restore it.
+
+### IC method `list_snapshots` {#ic-take_snapshot}
+
+This method lists the snapshots of the canister identified by `canister_id`. Currently, only one snapshot per canister will be stored.
+
+### IC method `delete_snapshot` {#ic-take_snapshot}
+
+This method deletes the snapshot of the canister. An error will be returned if the snapshot is not found. A snapshot cannot be 
+found if it was previously deleted, replaced by a new snapshot through a `take_snapshot` request, or if the canister itself has been deleted.
```

Comment:
> replaced by a new snapshot through a `take_snapshot` request

it sounds weird that the API is formulated as if multiple snapshots per canister were allowed but there's also the implicit replacement of an old snapshot if `take_snapshot` is called - wouldn't it make more sense to fail `take_snapshot` if there already exists a snapshot (forcing the user to call `delete_snapshot` before calling `take_snapshot` again)? I guess this is how it's gonna work once the 1-snapshot restriction is lifted in the future (if it's not gonna be lifted, then we should have vectors and plural in the API).

Reply:
cc @AlexandraZapuc

Reply:
> it sounds weird that the API is formulated as if multiple snapshots per canister were allowed  

@mraszyk  Which part is formulated as multiple snapshots?  Do you mean this :  `list_snapshots: (record {canister_id : canister_id}) -> (vec snapshot)`?

Reply:
I have changed the api following the discussion with @mraszyk. I will add the rest of the semantics.